earth_enterprise/src/fusion/autoingest/AssetHandle.h

```diff
@@ -149,11 +149,11 @@ class DerivedAssetHandle_ : public virtual Base_ {
 
  public:
   virtual HandleType Load(const std::string &boundref) const {
-    // Impl::Load will succeed or throw.
-    // The derived khRefGuard will be automatically converted
-    // the the base khRefGuard
-    return HandleType(Impl::Load(boundref));
+    // when switching to using shared_ptr, manually call Base load
+    // and don't rely on auto conversion, as is done via khRefGuard
+    return HandleType(Base::Load(boundref));
```

Comment:
This makes me a bit nervous - are we sure this has the same behavior as the previous version?

Reply:
it seems to, but tracing a bit more, it would probably be better to use Impl